src/actions/DeviceSettingsActions.ts

```diff
@@ -0,0 +1,42 @@
+import { makeReactNativeDisklet } from 'disklet'
+
+import { asDeviceSettings, DeviceSettings } from '../types/types'
+
+const disklet = makeReactNativeDisklet()
+const DEVICE_SETTINGS_FILENAME = 'DeviceSettings.json'
+let deviceSettings: DeviceSettings = asDeviceSettings({})
+
+export const getDeviceSettings = (): DeviceSettings => deviceSettings
+export const initDeviceSettings = async () => {
+  deviceSettings = await readDeviceSettings()
+}
+
+export const writeDisableAnimations = async (disableAnimations: boolean) => {
+  try {
+    const raw = await disklet.getText(DEVICE_SETTINGS_FILENAME)
+    const json = JSON.parse(raw)
+    deviceSettings = asDeviceSettings(json)
+  } catch (e) {
+    console.log(e)
+  }
+  const updatedSettings: DeviceSettings = { ...deviceSettings, disableAnimations }
+  return await writeDeviceSettings(updatedSettings)
+}
+
+const readDeviceSettings = async (): Promise<DeviceSettings> => {
+  try {
+    const text = await disklet.getText(DEVICE_SETTINGS_FILENAME)
+    const json = JSON.parse(text)
+    const settings = asDeviceSettings(json)
+    return settings
+  } catch (e) {
+    const defaults = asDeviceSettings({})
+    return await writeDeviceSettings(defaults).then(() => defaults)
```

Comment:
This isn't needed. Just return `{}` and be done with it. The file is only interesting once the user changes an actual setting.

Having an empty file on disk doesn't change anything or help anything - if it's not there, you return the defaults, and if the file is there, it only contains defaults anyhow, so why bother?